plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -225,6 +358,34 @@ static int init_put_payload(struct flb_cloudwatch *ctx, struct cw_flush *buf,
                       "\",", 2)) {
         goto error;
     }
+    // If we are missing the service name, the entity will get rejected by the frontend anyway
+    // so do not emit entity unless service name is filled
+    if(ctx->add_entity && stream->entity != NULL && stream->entity->key_attributes->name != NULL) {
```

Comment:
Do we know that `stream->entity->key_attributes` can never be NULL?

Reply:
addressed in new commit. adding NULL check here just in case.